### ISSUE TITLE
Couldn't run the example code

### ISSUE THREAD

**Issue by OoooL19** [borglab/gtsam#801]
I tried to run ```make CameraResectioning```, but I got this. I am new to gtsam. I try to learn the gtsam from the example. Can anyone help me with this issue? I am on ubuntu 18 and using eigen 3.3.9.  I use developer branch of the gtsam. Thank you.
```
g++     CameraResectioning.cpp   -o CameraResectioning
/tmp/ccKitNic.o: In function `main':
CameraResectioning.cpp:(.text+0x18c): undefined reference to `gtsam::noiseModel::Diagonal::Sigmas(Eigen::Matrix<double, -1, 1, 0, -1, 1> const&, bool)'
CameraResectioning.cpp:(.text+0x59f): undefined reference to `gtsam::Rot3::Rot3(double, double, double, double, double, double, double, double, double)'
CameraResectioning.cpp:(.text+0x63a): undefined reference to `gtsam::LevenbergMarquardtOptimizer::LevenbergMarquardtOptimizer(gtsam::NonlinearFactorGraph const&, gtsam::Values const&, gtsam::LevenbergMarquardtParams const&)'
CameraResectioning.cpp:(.text+0x65e): undefined reference to `gtsam::Values::Values(gtsam::Values const&)'
CameraResectioning.cpp:(.text+0x6c8): undefined reference to `gtsam::Values::print(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::function<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > (unsigned long)> const&) const'
/tmp/ccKitNic.o: In function `__static_initialization_and_destruction_0(int, int)':
CameraResectioning.cpp:(.text+0x942): undefined reference to `gtsam::_defaultKeyFormatter[abi:cxx11](unsigned long)'
CameraResectioning.cpp:(.text+0x972): undefined reference to `gtsam::_multirobotKeyFormatter[abi:cxx11](unsigned long)'
/tmp/ccKitNic.o: In function `gtsam::Symbol::operator unsigned long() const':
CameraResectioning.cpp:(.text._ZNK5gtsam6SymbolcvmEv[_ZNK5gtsam6SymbolcvmEv]+0x14): undefined reference to `gtsam::Symbol::key() const'
/tmp/ccKitNic.o: In function `gtsam::ValuesKeyDoesNotExist::ValuesKeyDoesNotExist(char const*, unsigned long)':
CameraResectioning.cpp:(.text._ZN5gtsam21ValuesKeyDoesNotExistC2EPKcm[_ZN5gtsam21ValuesKeyDoesNotExistC5EPKcm]+0x23): undefined reference to `vtable for gtsam::ValuesKeyDoesNotExist'
/tmp/ccKitNic.o: In function `gtsam::ValuesKeyDoesNotExist::~ValuesKeyDoesNotExist()':
CameraResectioning.cpp:(.text._ZN5gtsam21ValuesKeyDoesNotExistD2Ev[_ZN5gtsam21ValuesKeyDoesNotExistD5Ev]+0xf): undefined reference to `vtable for gtsam::ValuesKeyDoesNotExist'
/tmp/ccKitNic.o: In function `gtsam::ValuesIncorrectType::ValuesIncorrectType(unsigned long, std::type_info const&, std::type_info const&)':
CameraResectioning.cpp:(.text._ZN5gtsam19ValuesIncorrectTypeC2EmRKSt9type_infoS3_[_ZN5gtsam19ValuesIncorrectTypeC5EmRKSt9type_infoS3_]+0x27): undefined reference to `vtable for gtsam::ValuesIncorrectType'
/tmp/ccKitNic.o: In function `gtsam::ValuesIncorrectType::~ValuesIncorrectType()':
CameraResectioning.cpp:(.text._ZN5gtsam19ValuesIncorrectTypeD2Ev[_ZN5gtsam19ValuesIncorrectTypeD5Ev]+0xf): undefined reference to `vtable for gtsam::ValuesIncorrectType'
/tmp/ccKitNic.o: In function `gtsam::NonlinearFactor::~NonlinearFactor()':
CameraResectioning.cpp:(.text._ZN5gtsam15NonlinearFactorD2Ev[_ZN5gtsam15NonlinearFactorD5Ev]+0xf): undefined reference to `vtable for gtsam::NonlinearFactor'
/tmp/ccKitNic.o: In function `gtsam::NoiseModelFactor::~NoiseModelFactor()':
CameraResectioning.cpp:(.text._ZN5gtsam16NoiseModelFactorD2Ev[_ZN5gtsam16NoiseModelFactorD5Ev]+0xf): undefined reference to `vtable for gtsam::NoiseModelFactor'
/tmp/ccKitNic.o: In function `gtsam::NonlinearOptimizerParams::~NonlinearOptimizerParams()':
CameraResectioning.cpp:(.text._ZN5gtsam24NonlinearOptimizerParamsD2Ev[_ZN5gtsam24NonlinearOptimizerParamsD5Ev]+0xf): undefined reference to `vtable for gtsam::NonlinearOptimizerParams'
/tmp/ccKitNic.o: In function `gtsam::NonlinearOptimizer::optimize()':
CameraResectioning.cpp:(.text._ZN5gtsam18NonlinearOptimizer8optimizeEv[_ZN5gtsam18NonlinearOptimizer8optimizeEv]+0x14): undefined reference to `gtsam::NonlinearOptimizer::defaultOptimize()'
CameraResectioning.cpp:(.text._ZN5gtsam18NonlinearOptimizer8optimizeEv[_ZN5gtsam18NonlinearOptimizer8optimizeEv]+0x20): undefined reference to `gtsam::NonlinearOptimizer::values() const'
/tmp/ccKitNic.o: In function `gtsam::NonlinearOptimizerParams::NonlinearOptimizerParams()':
CameraResectioning.cpp:(.text._ZN5gtsam24NonlinearOptimizerParamsC2Ev[_ZN5gtsam24NonlinearOptimizerParamsC5Ev]+0xf): undefined reference to `vtable for gtsam::NonlinearOptimizerParams'
/tmp/ccKitNic.o: In function `gtsam::LevenbergMarquardtParams::LevenbergMarquardtParams()':
CameraResectioning.cpp:(.text._ZN5gtsam24LevenbergMarquardtParamsC2Ev[_ZN5gtsam24LevenbergMarquardtParamsC5Ev]+0x1b): undefined reference to `vtable for gtsam::LevenbergMarquardtParams'
/tmp/ccKitNic.o: In function `gtsam::LevenbergMarquardtParams::~LevenbergMarquardtParams()':
CameraResectioning.cpp:(.text._ZN5gtsam24LevenbergMarquardtParamsD2Ev[_ZN5gtsam24LevenbergMarquardtParamsD5Ev]+0xf): undefined reference to `vtable for gtsam::LevenbergMarquardtParams'
/tmp/ccKitNic.o: In function `gtsam::LevenbergMarquardtOptimizer::~LevenbergMarquardtOptimizer()':
CameraResectioning.cpp:(.text._ZN5gtsam27LevenbergMarquardtOptimizerD2Ev[_ZN5gtsam27LevenbergMarquardtOptimizerD5Ev]+0xf): undefined reference to `vtable for gtsam::LevenbergMarquardtOptimizer'
CameraResectioning.cpp:(.text._ZN5gtsam27LevenbergMarquardtOptimizerD2Ev[_ZN5gtsam27LevenbergMarquardtOptimizerD5Ev]+0x32): undefined reference to `gtsam::NonlinearOptimizer::~NonlinearOptimizer()'
/tmp/ccKitNic.o: In function `gtsam::Pose3::operator*(gtsam::Pose3 const&) const':
CameraResectioning.cpp:(.text._ZNK5gtsam5Pose3mlERKS0_[_ZNK5gtsam5Pose3mlERKS0_]+0x50): undefined reference to `gtsam::Rot3::operator*(Eigen::Matrix<double, 3, 1, 0, 3, 1> const&) const'
CameraResectioning.cpp:(.text._ZNK5gtsam5Pose3mlERKS0_[_ZNK5gtsam5Pose3mlERKS0_]+0xaa): undefined reference to `gtsam::Rot3::operator*(gtsam::Rot3 const&) const'
/tmp/ccKitNic.o: In function `gtsam::Cal3::Cal3(double, double, double, double, double)':
CameraResectioning.cpp:(.text._ZN5gtsam4Cal3C2Eddddd[_ZN5gtsam4Cal3C5Eddddd]+0x24): undefined reference to `vtable for gtsam::Cal3'
/tmp/ccKitNic.o: In function `gtsam::Cal3::~Cal3()':
CameraResectioning.cpp:(.text._ZN5gtsam4Cal3D2Ev[_ZN5gtsam4Cal3D5Ev]+0xb): undefined reference to `vtable for gtsam::Cal3'
/tmp/ccKitNic.o: In function `gtsam::Cal3_S2::Cal3_S2(double, double, double, double, double)':
CameraResectioning.cpp:(.text._ZN5gtsam7Cal3_S2C2Eddddd[_ZN5gtsam7Cal3_S2C5Eddddd]+0x65): undefined reference to `vtable for gtsam::Cal3_S2'
/tmp/ccKitNic.o: In function `gtsam::Cal3::Cal3(gtsam::Cal3 const&)':
CameraResectioning.cpp:(.text._ZN5gtsam4Cal3C2ERKS0_[_ZN5gtsam4Cal3C5ERKS0_]+0xf): undefined reference to `vtable for gtsam::Cal3'
/tmp/ccKitNic.o: In function `gtsam::Cal3_S2::Cal3_S2(gtsam::Cal3_S2 const&)':
CameraResectioning.cpp:(.text._ZN5gtsam7Cal3_S2C2ERKS0_[_ZN5gtsam7Cal3_S2C5ERKS0_]+0x26): undefined reference to `vtable for gtsam::Cal3_S2'
/tmp/ccKitNic.o: In function `void gtsam::Values::insert<gtsam::Pose3>(unsigned long, gtsam::Pose3 const&)':
CameraResectioning.cpp:(.text._ZN5gtsam6Values6insertINS_5Pose3EEEvmRKT_[_ZN5gtsam6Values6insertINS_5Pose3EEEvmRKT_]+0x65): undefined reference to `gtsam::Values::insert(unsigned long, gtsam::Value const&)'
/tmp/ccKitNic.o: In function `gtsam::NoiseModelFactor::NoiseModelFactor<boost::assign_detail::static_generic_list<unsigned long const, 1> >(boost::shared_ptr<gtsam::noiseModel::Base> const&, boost::assign_detail::static_generic_list<unsigned long const, 1> const&)':
CameraResectioning.cpp:(.text._ZN5gtsam16NoiseModelFactorC2IN5boost13assign_detail19static_generic_listIKmLi1EEEEERKNS2_10shared_ptrINS_10noiseModel4BaseEEERKT_[_ZN5gtsam16NoiseModelFactorC5IN5boost13assign_detail19static_generic_listIKmLi1EEEEERKNS2_10shared_ptrINS_10noiseModel4BaseEEERKT_]+0x2a): undefined reference to `vtable for gtsam::NoiseModelFactor'
/tmp/ccKitNic.o: In function `Eigen::Matrix<double, 2, 1, 0, 2, 1> gtsam::PinholeBaseK<gtsam::Cal3_S2>::_project<Eigen::Matrix<double, 3, 1, 0, 3, 1> >(Eigen::Matrix<double, 3, 1, 0, 3, 1> const&, gtsam::OptionalJacobian<2, 6>, gtsam::OptionalJacobian<2, gtsam::FixedDimension<Eigen::Matrix<double, 3, 1, 0, 3, 1> >::value>, gtsam::OptionalJacobian<2, 5>) const':
CameraResectioning.cpp:(.text._ZNK5gtsam12PinholeBaseKINS_7Cal3_S2EE8_projectIN5Eigen6MatrixIdLi3ELi1ELi0ELi3ELi1EEEEENS5_IdLi2ELi1ELi0ELi2ELi1EEERKT_NS_16OptionalJacobianILi2ELi6EEENSB_ILi2EXsrNS_14FixedDimensionIS8_EE5valueEEENSB_ILi2ELi5EEE[_ZNK5gtsam12PinholeBaseKINS_7Cal3_S2EE8_projectIN5Eigen6MatrixIdLi3ELi1ELi0ELi3ELi1EEEEENS5_IdLi2ELi1ELi0ELi2ELi1EEERKT_NS_16OptionalJacobianILi2ELi6EEENSB_ILi2EXsrNS_14FixedDimensionIS8_EE5valueEEENSB_ILi2ELi5EEE]+0x99): undefined reference to `gtsam::PinholeBase::project2(Eigen::Matrix<double, 3, 1, 0, 3, 1> const&, gtsam::OptionalJacobian<2, 6>, gtsam::OptionalJacobian<2, 3>) const'
CameraResectioning.cpp:(.text._ZNK5gtsam12PinholeBaseKINS_7Cal3_S2EE8_projectIN5Eigen6MatrixIdLi3ELi1ELi0ELi3ELi1EEEEENS5_IdLi2ELi1ELi0ELi2ELi1EEERKT_NS_16OptionalJacobianILi2ELi6EEENSB_ILi2EXsrNS_14FixedDimensionIS8_EE5valueEEENSB_ILi2ELi5EEE[_ZNK5gtsam12PinholeBaseKINS_7Cal3_S2EE8_projectIN5Eigen6MatrixIdLi3ELi1ELi0ELi3ELi1EEEEENS5_IdLi2ELi1ELi0ELi2ELi1EEERKT_NS_16OptionalJacobianILi2ELi6EEENSB_ILi2EXsrNS_14FixedDimensionIS8_EE5valueEEENSB_ILi2ELi5EEE]+0x13c): undefined reference to `gtsam::Cal3_S2::uncalibrate(Eigen::Matrix<double, 2, 1, 0, 2, 1> const&, gtsam::OptionalJacobian<2, 5>, gtsam::OptionalJacobian<2, 2>) const'
/tmp/ccKitNic.o: In function `gtsam::NonlinearFactor::NonlinearFactor<boost::assign_detail::static_generic_list<unsigned long const, 1> >(boost::assign_detail::static_generic_list<unsigned long const, 1> const&)':
CameraResectioning.cpp:(.text._ZN5gtsam15NonlinearFactorC2IN5boost13assign_detail19static_generic_listIKmLi1EEEEERKT_[_ZN5gtsam15NonlinearFactorC5IN5boost13assign_detail19static_generic_listIKmLi1EEEEERKT_]+0x2a): undefined reference to `vtable for gtsam::NonlinearFactor'
/tmp/ccKitNic.o: In function `gtsam::Cal3_S2::~Cal3_S2()':
CameraResectioning.cpp:(.text._ZN5gtsam7Cal3_S2D2Ev[_ZN5gtsam7Cal3_S2D5Ev]+0xf): undefined reference to `vtable for gtsam::Cal3_S2'
/tmp/ccKitNic.o: In function `tbb::tbb_allocator<unsigned long>::deallocate(unsigned long*, unsigned long)':
CameraResectioning.cpp:(.text._ZN3tbb13tbb_allocatorImE10deallocateEPmm[_ZN3tbb13tbb_allocatorImE10deallocateEPmm]+0x1c): undefined reference to `tbb::internal::deallocate_via_handler_v3(void*)'
/tmp/ccKitNic.o: In function `tbb::tbb_allocator<boost::shared_ptr<gtsam::NonlinearFactor> >::deallocate(boost::shared_ptr<gtsam::NonlinearFactor>*, unsigned long)':
CameraResectioning.cpp:(.text._ZN3tbb13tbb_allocatorIN5boost10shared_ptrIN5gtsam15NonlinearFactorEEEE10deallocateEPS5_m[_ZN3tbb13tbb_allocatorIN5boost10shared_ptrIN5gtsam15NonlinearFactorEEEE10deallocateEPS5_m]+0x1c): undefined reference to `tbb::internal::deallocate_via_handler_v3(void*)'
/tmp/ccKitNic.o: In function `tbb::tbb_allocator<unsigned long>::allocate(unsigned long, void const*)':
CameraResectioning.cpp:(.text._ZN3tbb13tbb_allocatorImE8allocateEmPKv[_ZN3tbb13tbb_allocatorImE8allocateEmPKv]+0x20): undefined reference to `tbb::internal::allocate_via_handler_v3(unsigned long)'
/tmp/ccKitNic.o: In function `tbb::tbb_allocator<boost::shared_ptr<gtsam::NonlinearFactor> >::allocate(unsigned long, void const*)':
CameraResectioning.cpp:(.text._ZN3tbb13tbb_allocatorIN5boost10shared_ptrIN5gtsam15NonlinearFactorEEEE8allocateEmPKv[_ZN3tbb13tbb_allocatorIN5boost10shared_ptrIN5gtsam15NonlinearFactorEEEE8allocateEmPKv]+0x20): undefined reference to `tbb::internal::allocate_via_handler_v3(unsigned long)'
/tmp/ccKitNic.o:(.data.rel.ro._ZTV18ResectioningFactor[_ZTV18ResectioningFactor]+0x10): undefined reference to `gtsam::NoiseModelFactor::print(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::function<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > (unsigned long)> const&) const'
/tmp/ccKitNic.o:(.data.rel.ro._ZTV18ResectioningFactor[_ZTV18ResectioningFactor]+0x18): undefined reference to `gtsam::NoiseModelFactor::equals(gtsam::NonlinearFactor const&, double) const'
/tmp/ccKitNic.o:(.data.rel.ro._ZTV18ResectioningFactor[_ZTV18ResectioningFactor]+0x30): undefined reference to `gtsam::NoiseModelFactor::error(gtsam::Values const&) const'
/tmp/ccKitNic.o:(.data.rel.ro._ZTV18ResectioningFactor[_ZTV18ResectioningFactor]+0x48): undefined reference to `gtsam::NoiseModelFactor::linearize(gtsam::Values const&) const'
/tmp/ccKitNic.o:(.data.rel.ro._ZTVN5gtsam17NoiseModelFactor1INS_5Pose3EEE[_ZTVN5gtsam17NoiseModelFactor1INS_5Pose3EEE]+0x10): undefined reference to `gtsam::NoiseModelFactor::print(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::function<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > (unsigned long)> const&) const'
/tmp/ccKitNic.o:(.data.rel.ro._ZTVN5gtsam17NoiseModelFactor1INS_5Pose3EEE[_ZTVN5gtsam17NoiseModelFactor1INS_5Pose3EEE]+0x18): undefined reference to `gtsam::NoiseModelFactor::equals(gtsam::NonlinearFactor const&, double) const'
/tmp/ccKitNic.o:(.data.rel.ro._ZTVN5gtsam17NoiseModelFactor1INS_5Pose3EEE[_ZTVN5gtsam17NoiseModelFactor1INS_5Pose3EEE]+0x30): undefined reference to `gtsam::NoiseModelFactor::error(gtsam::Values const&) const'
/tmp/ccKitNic.o:(.data.rel.ro._ZTVN5gtsam17NoiseModelFactor1INS_5Pose3EEE[_ZTVN5gtsam17NoiseModelFactor1INS_5Pose3EEE]+0x48): undefined reference to `gtsam::NoiseModelFactor::linearize(gtsam::Values const&) const'
/tmp/ccKitNic.o:(.data.rel.ro._ZTIN5gtsam17NoiseModelFactor1INS_5Pose3EEE[_ZTIN5gtsam17NoiseModelFactor1INS_5Pose3EEE]+0x10): undefined reference to `typeinfo for gtsam::NoiseModelFactor'
/tmp/ccKitNic.o: In function `gtsam::GenericValue<gtsam::Pose3>::print(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const':
CameraResectioning.cpp:(.text._ZNK5gtsam12GenericValueINS_5Pose3EE5printERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE[_ZNK5gtsam12GenericValueINS_5Pose3EE5printERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE]+0x50): undefined reference to `gtsam::demangle[abi:cxx11](char const*)'
/tmp/ccKitNic.o: In function `gtsam::Testable<gtsam::Pose3>::Equals(gtsam::Pose3 const&, gtsam::Pose3 const&, double)':
CameraResectioning.cpp:(.text._ZN5gtsam8TestableINS_5Pose3EE6EqualsERKS1_S4_d[_ZN5gtsam8TestableINS_5Pose3EE6EqualsERKS1_S4_d]+0x31): undefined reference to `gtsam::Pose3::equals(gtsam::Pose3 const&, double) const'
/tmp/ccKitNic.o: In function `gtsam::Testable<gtsam::Pose3>::Print(gtsam::Pose3 const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
CameraResectioning.cpp:(.text._ZN5gtsam8TestableINS_5Pose3EE5PrintERKS1_RKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE[_ZN5gtsam8TestableINS_5Pose3EE5PrintERKS1_RKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE]+0x1f): undefined reference to `gtsam::Pose3::print(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const'
/tmp/ccKitNic.o: In function `gtsam::Pose3 const gtsam::Values::at<gtsam::Pose3>(unsigned long) const':
CameraResectioning.cpp:(.text._ZNK5gtsam6Values2atINS_5Pose3EEEKT_m[_ZNK5gtsam6Values2atINS_5Pose3EEEKT_m]+0x8f): undefined reference to `typeinfo for gtsam::ValuesKeyDoesNotExist'
/tmp/ccKitNic.o: In function `gtsam::LieGroup<gtsam::Pose3, 6>::retract(Eigen::Matrix<double, 6, 1, 0, 6, 1> const&, gtsam::OptionalJacobian<6, 6>, gtsam::OptionalJacobian<6, 6>) const':
CameraResectioning.cpp:(.text._ZNK5gtsam8LieGroupINS_5Pose3ELi6EE7retractERKN5Eigen6MatrixIdLi6ELi1ELi0ELi6ELi1EEENS_16OptionalJacobianILi6ELi6EEES9_[_ZNK5gtsam8LieGroupINS_5Pose3ELi6EE7retractERKN5Eigen6MatrixIdLi6ELi1ELi0ELi6ELi1EEENS_16OptionalJacobianILi6ELi6EEES9_]+0x9c): undefined reference to `gtsam::Pose3::ChartAtOrigin::Retract(Eigen::Matrix<double, 6, 1, 0, 6, 1> const&, gtsam::OptionalJacobian<6, 6>)'
CameraResectioning.cpp:(.text._ZNK5gtsam8LieGroupINS_5Pose3ELi6EE7retractERKN5Eigen6MatrixIdLi6ELi1ELi0ELi6ELi1EEENS_16OptionalJacobianILi6ELi6EEES9_[_ZNK5gtsam8LieGroupINS_5Pose3ELi6EE7retractERKN5Eigen6MatrixIdLi6ELi1ELi0ELi6ELi1EEENS_16OptionalJacobianILi6ELi6EEES9_]+0xe8): undefined reference to `gtsam::Pose3::inverse() const'
CameraResectioning.cpp:(.text._ZNK5gtsam8LieGroupINS_5Pose3ELi6EE7retractERKN5Eigen6MatrixIdLi6ELi1ELi0ELi6ELi1EEENS_16OptionalJacobianILi6ELi6EEES9_[_ZNK5gtsam8LieGroupINS_5Pose3ELi6EE7retractERKN5Eigen6MatrixIdLi6ELi1ELi0ELi6ELi1EEENS_16OptionalJacobianILi6ELi6EEES9_]+0x101): undefined reference to `gtsam::Pose3::AdjointMap() const'
/tmp/ccKitNic.o: In function `gtsam::LieGroup<gtsam::Pose3, 6>::localCoordinates(gtsam::Pose3 const&, gtsam::OptionalJacobian<6, 6>, gtsam::OptionalJacobian<6, 6>) const':
CameraResectioning.cpp:(.text._ZNK5gtsam8LieGroupINS_5Pose3ELi6EE16localCoordinatesERKS1_NS_16OptionalJacobianILi6ELi6EEES6_[_ZNK5gtsam8LieGroupINS_5Pose3ELi6EE16localCoordinatesERKS1_NS_16OptionalJacobianILi6ELi6EEES6_]+0xcf): undefined reference to `gtsam::Pose3::ChartAtOrigin::Local(gtsam::Pose3 const&, gtsam::OptionalJacobian<6, 6>)'
CameraResectioning.cpp:(.text._ZNK5gtsam8LieGroupINS_5Pose3ELi6EE16localCoordinatesERKS1_NS_16OptionalJacobianILi6ELi6EEES6_[_ZNK5gtsam8LieGroupINS_5Pose3ELi6EE16localCoordinatesERKS1_NS_16OptionalJacobianILi6ELi6EEES6_]+0xff): undefined reference to `gtsam::Pose3::inverse() const'
CameraResectioning.cpp:(.text._ZNK5gtsam8LieGroupINS_5Pose3ELi6EE16localCoordinatesERKS1_NS_16OptionalJacobianILi6ELi6EEES6_[_ZNK5gtsam8LieGroupINS_5Pose3ELi6EE16localCoordinatesERKS1_NS_16OptionalJacobianILi6ELi6EEES6_]+0x118): undefined reference to `gtsam::Pose3::AdjointMap() const'
/tmp/ccKitNic.o: In function `gtsam::internal::handle<gtsam::Pose3>::operator()(unsigned long, gtsam::Value const*)':
CameraResectioning.cpp:(.text._ZN5gtsam8internal6handleINS_5Pose3EEclEmPKNS_5ValueE[_ZN5gtsam8internal6handleINS_5Pose3EEclEmPKNS_5ValueE]+0xbc): undefined reference to `typeinfo for gtsam::ValuesIncorrectType'
/tmp/ccKitNic.o: In function `gtsam::LieGroup<gtsam::Pose3, 6>::between(gtsam::Pose3 const&) const':
CameraResectioning.cpp:(.text._ZNK5gtsam8LieGroupINS_5Pose3ELi6EE7betweenERKS1_[_ZNK5gtsam8LieGroupINS_5Pose3ELi6EE7betweenERKS1_]+0x4d): undefined reference to `gtsam::Pose3::inverse() const'
collect2: error: ld returned 1 exit status
<builtin>: recipe for target 'CameraResectioning' failed
make: *** [CameraResectioning] Error 1
```

**Comment by ProfFan** [comment]
Closed as non-issue.

Please follow the documentation on how to build examples properly **with CMake**. If you have further questions please use the mailing list.

**Comment by OoooL19** [comment]
> Closed as non-issue.
> 
> Please follow the documentation on how to build examples properly **with CMake**. If you have further questions please use the mailing list.

Yes, I did followed the example Readme, but I have this error. 
```
CMake Error at CMakeLists.txt:6 (gtsamAddExamplesGlob):
  Unknown CMake command "gtsamAddExamplesGlob".
```
When you said the use the mailing list. Did you mean email you guys? I am sorry I am also new to the github.

**Comment by ProfFan** [comment]
Please post your entire CMake command line and log so we can diagnose the issue.

**Comment by OoooL19** [comment]
> Please post your entire CMake command line and log so we can diagnose the issue.

In the examples folder I did
```
mkdir build
cd build
cmake ..
```
[CMakeOutput.log](https://github.com/borglab/gtsam/files/6690073/CMakeOutput.log)

**Comment by ProfFan** [comment]
This is not the CMake log, sorry but what I mean is the command and result that appeared on your terminal.
Please look at the Issue template for details.

**Comment by OoooL19** [comment]
> This is not the CMake log, sorry but what I mean is the command and result that appeared on your terminal.
> Please look at the Issue template for details.

I am sorry, these are the output. 
```
owen@owen-Terminator-T430:~/gtsam/examples$ mkdir build
owen@owen-Terminator-T430:~/gtsam/examples$ cd build
owen@owen-Terminator-T430:~/gtsam/examples/build$ cmake ..
CMake Warning (dev) in CMakeLists.txt:
  No project() command is present.  The top-level CMakeLists.txt file must
  contain a literal, direct call to the project() command.  Add a line of
  code such as

    project(ProjectName)

  near the top of the file, but after cmake_minimum_required().

  CMake is pretending there is a "project(Project)" command on the first
  line.
This warning is for project developers.  Use -Wno-dev to suppress it.

-- The C compiler identification is GNU 7.5.0
-- The CXX compiler identification is GNU 7.5.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at CMakeLists.txt:6 (gtsamAddExamplesGlob):
  Unknown CMake command "gtsamAddExamplesGlob".


-- Configuring incomplete, errors occurred!
See also "/home/owen/gtsam/examples/build/CMakeFiles/CMakeOutput.log".
```

**Comment by ProfFan** [comment]
This is not correct, but I think the documentation may need more clarification.

What you should do is run these commands under the *top* folder, not inside `example` folder.

```
mkdir build
cd build
cmake ..
```

We'll update the docs later.

**Comment by OoooL19** [comment]
> This is not correct, but I think the documentation may need more clarification.
> 
> What you should do is run these commands under the _top_ folder, not inside `example` folder.
> 
> ```
> mkdir build
> cd build
> cmake ..
> ```
> 
> We'll update the docs later.

Ok, In that case, I already did the build in the top folder before. Is there anything I missed, so I cannot do ``` make CameraResectioning ```

**Comment by ProfFan** [comment]
Again, please submit the console logs.

**Comment by OoooL19** [comment]
> Again, please submit the console logs.

yes, the first pose is the log I had when I ran the ```make CameraResectioning```
or you want the ```ccmake ..``` log?

**Comment by ProfFan** [comment]
1. remove everything
2. clone fresh repo
3. `mkdir build && cd build && cmake ..` (don't use ccmake if you don't understand CMake), logs
4. `make`, logs
5. `make CameraResectioning`, logs.

**Comment by OoooL19** [comment]
> 1. remove everything
> 2. clone fresh repo
> 3. `mkdir build && cd build && cmake ..` (don't use ccmake if you don't understand CMake), logs
> 4. `make`, logs
> 5. `make CameraResectioning`, logs.

This is the log I had when I tried to run ```make CameraResectioning``` after I did the previous steps.
```
owen@owen-Terminator-T430:~/gtsam/build$ make CameraResectioning
[ 30%] Built target metis-gtsam
[100%] Built target gtsam
[100%] Built target CameraResectioning
owen@owen-Terminator-T430:~/gtsam/build$ make CameraResectioning.run
[ 30%] Built target metis-gtsam
[100%] Built target gtsam
[100%] Built target CameraResectioning
./CameraResectioning: symbol lookup error: ./CameraResectioning: undefined symbol: _ZTVN5gtsam6FactorE
examples/CMakeFiles/CameraResectioning.run.dir/build.make:70: recipe for target 'examples/CMakeFiles/CameraResectioning.run' failed
make[3]: *** [examples/CMakeFiles/CameraResectioning.run] Error 127
CMakeFiles/Makefile2:20669: recipe for target 'examples/CMakeFiles/CameraResectioning.run.dir/all' failed
make[2]: *** [examples/CMakeFiles/CameraResectioning.run.dir/all] Error 2
CMakeFiles/Makefile2:20676: recipe for target 'examples/CMakeFiles/CameraResectioning.run.dir/rule' failed
make[1]: *** [examples/CMakeFiles/CameraResectioning.run.dir/rule] Error 2
Makefile:8792: recipe for target 'CameraResectioning.run' failed
make: *** [CameraResectioning.run] Error 2
```

**Comment by ProfFan** [comment]
You did not remove everything and start from clean. Make sure you post ALL logs from step one.

**Comment by OoooL19** [comment]
> You did not remove everything and start from clean. Make sure you post ALL logs from step one.

Hi, after I tried couple times. I finally make it works. Thank you so much.

**Comment by irisgro** [comment]
i meet the same question, could i know how did you solve this problem at last?
